src/inmanta/deploy/state.py

```diff
@@ -92,7 +92,7 @@ def __post_init__(self) -> None:
         object.__setattr__(self, "id", resources.Id.parse_id(self.resource_id))
 
 
-class DeploymentResult(StrEnum):
+class DeployStatus(StrEnum):
```

Comment:
rename 5.

motivation: conflict with `executor.DeployResult`. The executor one is aptly named imo.

notes:
- rename of `ResourceState.deployment_result` -> `ResourceState.deploy_status` will require db migration script

personal opinion: ambiguity improvement => make the change.

Reply:
I like that `result` makes clear this is about the outcome of a specific event, not some aggregated status. 
So I'm a bit on the fence about this one.

Reply:
For me, `executor.DeploymentResult` is very clearly the outcome of an action. This one slightly less so. But, now that I think about it, I do have to admit that it's very close.

I do think the name similarity remains confusing.

Reply:
I think we have to stress here that this represents in the outcome of the last deploy. So what about `StatusLastDeploy`?

Reply:
Or perhaps, combining your and @wouterdb's concerns, `LastDeployResult`? But, while I can see that for the field name (`last_deploy_result`), I don't feel like it belongs in the enum name. The fact that it represents the last deploy is a property of the `ResourceState` field, not of the enum itself, which represents nothing more than "the result of a deploy"

Reply:
I'm inclined to leave this one as is if we can't find something that we all agree is better. Optionally with a field rename to prefix it with `last_`.

But, that would mean that we are stuck with the name similarity between this and `executor.DeployResult`. And I think that one is definitely a good name. So, that's definitely unfortunate. But perhaps acceptable, assuming we use qualified imports (we don't everywhere atm), and the fact that we simply don't see anything that's better per se?

Reply:
Discussed in person, agreed on that latest proposal

Reply:
Discussed in person again: addendum: change `executor.DeployResult` -> `executor.DeployReport` and similar for the other resource action results/reports.